src/components/ChatSetting.vue

```diff
@@ -18,6 +25,77 @@ import ConfirmModal from "@/components/ConfirmModal.vue";
 const emit = defineEmits(["close-dialog"]);
 const confirmModal = ref();
 const store = useStore();
+
+// This function downloads the chat history as a JSON file.
+const downloadJson = () => {
+  // Get the chat history from localStorage.
+  const chatallMessages = localStorage.getItem("chatall-messages");
+  if (!chatallMessages) {
+    console.error("chatall-messages not found in localStorage");
+    return;
+  }
+
+  const chats = JSON.parse(chatallMessages)?.chats ?? [];
+
+  // Create an array of messages from the chat history.
+  const messages = chats
+    .filter((d) => !d.hide)
+    .map((chat) => ({
+      // The title of the chat.
+      title: chat.title,
+      // The messages in the chat.
+      messages: chat.messages
+        .filter((d) => !d.hide)
+        .reduce((arr, message) => {
+          const t = message.type;
+          const content = message.content;
+          if (t == "prompt") {
+            arr.push({
+              prompt: content,
+              responses: [],
+            });
+          } else {
+            arr.at(-1).responses.push({
+              content,
+              botClassname: message.className,
+              botModelName: message.model,
+              highlighted: message.highlight,
```

Comment:
```suggestion
              botClassname: message.className,
              botModel: message.model,
              highlight: message.highlight,
```

Moreover, we need a human readable name (Bing Chat (Precise) ) of the bot.

Reply:
What do you think of the following names? 
```
botName: message.className,
botModel: message.model,
emphasized: message.highlight,
```
`Bing Chat (Precise)` seems to be the agreed-upon name for the bot.

Reply:
`highlight` is ok. We must import new symbol upon good reason.

Reply:
What do I need to do now?